src/libs/ReportUtils.ts

```diff
@@ -946,7 +946,7 @@ function hasOnlyDistanceRequestTransactions(iouReportID: string | undefined): bo
  * If the report is a thread and has a chat type set, it is a workspace chat.
  */
 function isWorkspaceThread(report: OnyxEntry<Report>): boolean {
-    return isThread(report) && !isDM(report);
+    return isThread(report) && isChatReport(report) && !isDM(report);
```

Comment:
I have written this in my proposal.

The diagonal avatar pattern was missing from the Expense report and IOU report because there is a bug in isWorkspaceThread function which also gives true for the thread which is not a chat (i.e., IOU or expense).